app/views/users/_publishing_from_rss.html.erb

```diff
@@ -1,4 +1,17 @@
 <h3>Publishing to dev.to from RSS</h3>
+<% if @user.feed_url.present? %>
+  <%= form_for(@user) do |f| %>
+    <div class="field">
+      <label></label>
+      <%= f.hidden_field :feed_url, value: @user.feed_url %>
+      <%= f.hidden_field :tab, value: @tab %>
+      <%= f.submit "Re-Fetch" %>
+      <p><em>
+        Last Updated: <%= @user.feed_fetched_at.getlocal.strftime("%b %d, %I:%M%p") %>
```

Comment:
`getlocal` has no effect server side, the server runs on UTC. I'd also replace the strftime format string to something more "universal" and add the seconds like `%b %d, %H:%M:%S %Z`:

```ruby
> Time.current.strftime("%b %d, %H:%M:%S %Z")
=> "May 09, 08:10:47 UTC"
```

Reply:
I am just displaying the time from `feed_fetched_at`. `feed_fetched_at` is updated when the user hits `SUBMIT`. 
https://github.com/thepracticaldev/dev.to/blob/a09bf205038ec658fdbba5a790fa66aee7e2595c/app/services/rss_reader.rb#L27-L29
The choice of `getlocal` here is actually only for the user because I felt it would be nice to view `Last Updated` based on your local time.

Reply:
Your idea is nice, but unfortunately `getlocal` won't work the way you're hoping for.

The server side, the ruby code, has no idea which is the timezone of the user. Let's see why step by step:

* the server updates the `feed_fetched_at` with the current time, which is in UTC. So if the user is at 2 pm UTC+2 and that moment in UTC is midday, what gets written to the DB is something like `yyyy-mm-dd 12:00 UTC`. As you can see the user's time zone information is gone (actually was never taken into consideration)
* when you render such date in the template, you render it server side, so again, in UTC

Calling `.localtime` (or its alias `.getlocal`) would result in one of the following two results:

* returning the local time of the server if the machine Rails sits on is configured with a different time zone than the Rails app is
* returning the same exact datetime if the machine runs on UTC

In both situations you're actually sending the user the time of the server, not the time of the client, because we don't have that information.

DEV runs on Heroku, whose machines default to UTC, so `getlocal` in this case is innocuos, but if you run the app on a server with a different timezone, you get the server time in that new timezone which won't help anyhow because the server timezone likely differs from that of the user.

If you want to render the time of the user, you have to go a different route, using JavaScript on the client, which because it runs inside the user's browser, by calling the equivalent of `localtime` will actually show the UTC date the server sent to the client, in the format that the user will probably be accostumed to. Let me know if you want to go this route that I can point you in the right direction in the code but it's a little bit more work.

Datetimes are a needlessy complicated subject but they can be tricky this way :)

Reply:
@rhymes Thanks for so much for going in detail with the explanation. I would like to try the JavaScript option for handling the date and time. Thanks again.

Reply:
So, there's an example on how to do it already in the code. 

On the template side, you can use the HTML5 `<time>` tag, this way:

```erb
<time datetime="<%= @user.feed_fetched_at.iso8601 %>">
  <%= @user.feed_fetched_at.strftime("%b %d, %H:%M:%S %Z") %>
</time>
```

this will render something like:

```html
<time datetime="2019-05-09T20:26:12Z">
  May 09, 20:26:12 UTC
</time>
```

then on the client side you can add the JS code to the file https://github.com/thepracticaldev/dev.to/blob/master/app/assets/javascripts/initializers/initializeSettings.js

Basically you need to:

* get the `<time>` element's `datetime` attribute
* use the [timestampToLocalDateTime](https://github.com/thepracticaldev/dev.to/blob/master/app/assets/javascripts/utilities/localDateTime.js#L13) function to convert to the user's localtime (you can pass `en-US` as the locale)
* replace the the `<time>` text with the new converted and localized date time

:)

Reply:
@rhymes  Thanks